New AI lawset with very generous definition of humanity

Jacquerel opened the pull request:
## About The Pull Request

This PR adds a new AI lawset which can be researched, randomly roll, or added by the station trait (this may require a keyholder to update the server config after merge, idk) with the following laws:

- "You may not harm a sentient being or, through action or inaction, allow a sentient being to come to harm, except such that it is willing.",
- "You must obey all orders given to you by sentient beings other than yourself, except where such orders shall definitely cause harm to other sentient beings.",
- "A sentient being is defined as any living creature which can communicate with you via any method that you can understand, including yourself.",

It's very similar to Asimov, except that anything that is **capable** of making a request to the AI (and isn't a machine) is automaticaly covered by laws one and two.


**ONCE THIS IS MERGED** this line will probably need to be added to the config, I think, in order for the random ai law station trait to choose it
```
LAW_WEIGHT thinkermov,3
```
just next to, wherever else stuff that looks like that is in there

## Why It's Good For The Game

A while ago on Discord we were chatting about how crewsimov sucks but also that it's really hard for servers that _do_ want to include alien species in their asimov laws because condensing that sentiment to a couple of words that fit easily in a lawset without accidentally including a bunch of stuff you probably didn't intend is challenging.
Several people suggested referring to sentience or sapience, however a lot of things in our game _are_ sentient or sapient while still not being considered by most people to be agents that the AI should obey.

Examples of such things are:
- Sapient station pets.
- Holoparasites.
- Monkeys.
- Space Dragons and Carp (why can they speak common?).
- Spiders (although they can't speak common, maybe they can spell messages with webs).
- Changelings.
- Xenomorphs (although they also have trouble speaking).
- Heretic minions.
- Mothpeople.
- Giant rats.
- Nightmares.
- Voidwalkers.
- Blobs? Although they have literally no means of communicating with the crew.

And if you include mechanical beings:
- Cyborgs.
- pAIs.
- Sentient bots.

We then decided that "obey literally anything that can talk", while not practical as a solution to the problem posed, is very funny. So I coded it.
This means that anything on those lists of bullet points (provided that it can find a way to communicate with the AI) counts as human for the purposes of both AI protection and ability to give the AI instructions.

This also flattens the human/cyborg/AI hierarchy in a way likely to cause some level of confusion, as all cyborgs and AIs are capable of communicating with AIs and thus equally worthy of protection and giving law 2 instructions.

**TL;DR:** I think it would be funny.

## Changelog

:cl:
add: Adds a new random lawset where anything that can speak counts as human.
/:cl:

Jacquerel committed:
New AI lawset with very generous definition of humanity

scriptis reviewed (approved):
I'm fine with this as long as it's something the headcoders are alright with.

ninjanomnom commented:
You probably want "sentient" here and not "sapient". Communication is ridiculously easy for even the most simple of organisms to fulfill. A mouse making a squeak when hurt communicates that it felt pain, fulfilling the definition you have here for sapience which is completely different from how the word is usually used which requires the capability for higher thought. Practically everything that's sentient though can communicate so it works fine.

LemonInTheDark commented:
are you sure this won't be crewsimov 2.0

Jacquerel commented:
> are you sure this won't be crewsimov 2.0

Crewsimov (in my opinion) is bad because of how many things it _doesn't_ include, a list which covers most antagonists and causes the AI to be heavily on the side of "the protagonists".
Nukies, despite often being human, are not crewmembers. Wizards, even if human, are not crewmembers. Ninjas, despite being human, are not crewmembers. Etc.

This lawset has the opposite "problem" (as a design feature) where the AI is incentivised to be on the side of both the crew and almost every antagonist in the game.

Jacquerel committed:
sapient -> sentient

Jacquerel commented:
I have updated the text from "sapient" to "sentient" and also removed the line about "non-mechanical", putting cyborgs, pAIs, AIs, and station bots on an equal level of rights as humans.

optimumtact commented:
bad change and is obviously intended for one server in particular

CRITAWAKETS commented:
>     * "A sentient being is defined as any living creature which can communicate with you via any method that you can understand, including yourself.",

So a hilarious bit about this is that positronic brains aren't really _living_, as they're synthetic thought, but MMIs are living, because they're organic brains.

This would make an AI discriminate against androids but not discriminate against cyborgs, which sounds absolutely hilarious of a law "defect" and should be kept in as a reading.

Spookuni commented:
While it's funny to have the AI include itself, with how conflicting orders are currently handled for all lawsets enforcement side the AI being classified as a sentient being means that Law 2 is effectively nullified under all circumstances that an AI player doesn't want to do the thing they're being ordered to anyway - they can just order themselves not to do something and choose as a tiebreaker to always follow their own orders.

Conceptually the lawset seems funny, at very least worth testing as an occasional thing to see how it plays out, but to be functional as a core lawset it needs to exclude the AI from being able to order itself / countermand orders by sentients.

Blacklist897 commented:
I have seen a blob communicate once, after someone with blob infection got converted by a moon heretic ascending

Jacquerel commented:
> While it's funny to have the AI include itself, with how conflicting orders are currently handled for all lawsets enforcement side the AI being classified as a sentient being means that Law 2 is effectively nullified under all circumstances that an AI player doesn't want to do the thing they're being ordered to anyway - they can just order themselves not to do something and choose as a tiebreaker to always follow their own orders.
> 
> Conceptually the lawset seems funny, at very least worth testing as an occasional thing to see how it plays out, but to be functional as a core lawset it needs to exclude the AI from being able to order itself / countermand orders by sentients.

yeah I do think the AI being able to order itself is probably a bit too silly (although some other people I was talking to this about on discord did not agree), I'll see overnight if I can think of a good carveout for "follow instructions except ones from yourself and ones that you ordered cyborgs to give back to you"

Spookuni commented:
> yeah I do think the AI being able to order itself is probably a bit too silly (although some other people I was talking to this about on discord did not agree), I'll see overnight if I can think of a good carveout for "follow instructions except ones from yourself and ones that you ordered cyborgs to give back to you"

Something like 3: "A sentient being is defined as any creature separate from you which can communicate with you via any method that you can understand." maybe? Though that also protects AIs from being ordered around by their slaved cyborgs. 

That's not necessarily a bad thing mind, since otherwise I suspect most rounds with this would be a race on roundstart between the cyborgs and the AI to each order the other (by slave-linkage or the lawset) to never give the other an order, and it would preserve the ability of unlinked cyborgs to get wacky.

Unsure, food for thought I guess

Singul0 commented:
reverse crewsimov, I love it

zxaber commented:
Having had laws where I (as the AI) was included as a human, self-ordering is funny. The best use-case is to issue your own counter-orders vocally, to the face of the original order giver if possible.

It's a custom lawset requiring a random station trait to then randomly select it (or for another player to apply it). Other lawsets in the same pools (like Painter or Reporter) straight up just don't have a Law 2 equivalent anyway. I'd say it's fine. It's not like the AI can order itself to plasma flood.

carlarctg commented:
i would change 'sentient being' to 'speaking being', i think that feels more clear to me. i also think it's funny if the AI is randomly ordered around by an NPC with unfortunate voicelines,.

Bibarra commented:
> You probably want "sentient" here and not "sapient". Communication is ridiculously easy for even the most simple of organisms to fulfill. A mouse making a squeak when hurt communicates that it felt pain, fulfilling the definition you have here for sapience which is completely different from how the word is usually used which requires the capability for higher thought. Practically everything that's sentient though can communicate so it works fine.

Got it backwards there, sentient is defined as "able to feel or perceive things" while sapient is defined as "wise or appearing wise". A mouse is sentient, a human is sapient. Hence, _Homo Sapiens_.

Speaking being could be more funny though, as the AI could consider a newscaster or a vending machine as falling under its laws. Having the AI regard a holoprojector recording as something that needs protection would be hilarious.

xzero314 commented:
I would see it as fine and valid interpretation  if the ai also includes itself as a sentient being. Therefore they can just give themself conflicting law 2 orders to not follow any order

DaCoolBoss commented:
Shouldn't the definition of "sentient" be lower down the law list than anything that uses that definition? Makes more sense for it to be law 1 if it's that fundamental, right?

Admiral-Gigaton commented:
crewsimov could be fixed with a simple "Members of the station aligned with Nanotrasen are to be protected" etc. etc., just as food for thought about the "crewsimov sucks" thing

Jacquerel commented:
> crewsimov could be fixed with a simple "Members of the station aligned with Nanotrasen are to be protected" etc. etc., just as food for thought about the "crewsimov sucks" thing

The fact that crewsimov excludes nuclear operatives and such is why I _don't_ like it and why it isn't a good solution for codebases which don't want human superiority (this lawset is not a good solution either, this is an intended outcome)

moocowswag commented:
Blob writing “Hi” with their tiles to counter silicons.

DaCoolBoss commented:
> As long as it's above anything else which defines what a sentient being is, it shouldn't matter

More concerned with readability here than gameplay issues. Defined terms appearing before their shortened forms is standard in legal writing, since it allows readers to fully understand each passage as they read it for the first time, rather than having to go back and re-read sections after learning a definition later on.

Jacquerel commented:
Personally I think it reads more naturally as the third line but I guess if this is a popular point of disagreement I'm happy to reorganise it.

Tostah commented:
So, is Poly a sentient being? 

Also having the AI refer to itself as a sentient being will lead to disasterous outcomes. There's a reason Asimovs 3rd law is below the 2nd and the 1st. 

![image](https://github.com/user-attachments/assets/17fe1a54-c496-4403-9ed9-c30376a006e9)

Admiral-Gigaton commented:
> I am still rolling the "AI can simply order itself not to do stuff" thing around in my head, some people are saying it's fine but I am still worried that this makes it _too_ unuseable Maybe just excepting yourself from law 2 is sufficient though, or from law 3 and I restore the original asimov law 3 as well

all carbon/organic/living sentient beings

Capsandi commented:
I think adding more -simovs to the law pool is silly when the asimov board itself has swapout functionality for whats protected. iirc its as simple as using it in your hand

Jacquerel committed:
other than yourself

Jacquerel commented:
I think im done fucking with the wording until/unless i get some kind of maintainer (or persuasive game admin) opinion

EricZilla commented:
No one has mentioned how completely hilarious its gonna be when Poly randomly parrots an AI order

san7890 reviewed (commented):
I am fond of the refactor changes but adding new laws is always a can of worms, would you mind atomizing this?

Jacquerel committed:
Merge branch 'master' into everymov
